xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -15002,20 +15002,20 @@ glusterd_add_peers_to_auth_list(char *volname)
          * keep the copy of auth_allow_list as old_auth_allow_list in
          * volinfo->dict.
          */
-        dict_del_sizen(volinfo->dict, "auth.allow");
-        ret = dict_set_strn(volinfo->dict, "auth.allow", SLEN("auth.allow"),
-                            new_auth_allow_list);
+        ret = dict_set_dynstr_with_alloc(volinfo->dict, "old.auth.allow",
+                                         auth_allow_list);
         if (ret) {
             gf_msg(this->name, GF_LOG_ERROR, -ret, GD_MSG_DICT_SET_FAILED,
-                   "Unable to set new auth.allow list");
+                   "Unable to set old.auth.allow list");
             goto out;
         }
-        ret = dict_set_strn(volinfo->dict, "old.auth.allow",
-                            SLEN("old.auth.allow"), auth_allow_list);
+        ret = dict_set_dynstr(volinfo->dict, "auth.allow", new_auth_allow_list);
         if (ret) {
             gf_msg(this->name, GF_LOG_ERROR, -ret, GD_MSG_DICT_SET_FAILED,
-                   "Unable to set old auth.allow list");
+                   "Unable to set new auth.allow list");
             goto out;
+        } else {
```

Comment:
This is not technically necessary since the previous `if` always jumps to `out` label, but it's not a blocker either and the rest of the patch seems fine to me. Feel free to modify it or merge the patch, as you prefer.

Reply:
done